pandas/core/indexes/range.py

```diff
@@ -64,6 +65,8 @@ class RangeIndex(Int64Index):
     _typ = 'rangeindex'
     _engine_type = libindex.Int64Engine
 
+    # check whether self._data has benn called
+    _has_called_data = False  # type: bool
```

Comment:
This is added to check if ``._data`` has been called, without actually calling it..